### PR TITLE
Add STM32U0 support.

### PR DESCRIPTION
**Pull request by Dirbaio**
- **stm32/spi,crc: update for new PAC**
- **stm32/flash: remove otp flash regions (removed in newer metapacs).**
- **stm32/adc: update g4 for new pac.**
- **Add stm32u0 support.**

### CODE REVIEW

**Comment by Dirbaio** [comment]
bender run

**Comment by Dirbaio** [comment]
bender run

**Comment by Dirbaio** [comment]
bender run

File: examples/stm32u0/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[target.'cfg(all(target_arch = "arm", target_os = "none"))']
+# replace STM32G071C8Rx with your chip as listed in `probe-rs chip list`
+runner = "probe-rs run --speed 100 --chip STM32u083rctx"
```
Comment (diondokter) [inline examples/stm32u0/.cargo/config.toml:3]:
```suggestion
runner = "probe-rs run --chip STM32u083rctx"
```

Not sure if you found you needed to limit the speed, but it works fine for me without (and it's almost 10x the speed)

Comment (Dirbaio) [reply examples/stm32u0/.cargo/config.toml:3]: ahh tha'ts why it flashed so slow! :rofl:  (it's copypasted from the c0 examples...)

**Pull request merged**